dkpro-core-lancaster-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/lancaster/LancasterStemmer.java

```diff
@@ -0,0 +1,144 @@
+
+    /**
+     * True if the stemmer will strip prefix such as kilo, micro, milli, intra, ultra, mega, nano, pico, pseudo.
+     */
+    public static final String PARAM_STRIP_PREFIXES = "stripPrefix";
+    @ConfigurationParameter(name = PARAM_STRIP_PREFIXES, mandatory = false, defaultValue="false")
+    protected boolean stripPrefix;
+
+    @Override
+    protected Set<String> getDefaultPaths()
+    {
+        return Collections.singleton(Token.class.getName());
+    }
+
+    @Override
+    protected void generateAnnotations(JCas jcas) throws FeaturePathException, AnalysisEngineProcessException {
+
+            // CAS is necessary to retrieve values
+            CAS currCAS = jcas.getCas();
+
+            // Try language set in CAS.
+      		String lang = jcas.getDocumentLanguage();
+
+      		if (StringUtils.isBlank(lang)) {
+      			throw new AnalysisEngineProcessException(MESSAGE_DIGEST, "no_language_error", null);
+      		}
+
+      		lang = lang.toLowerCase(Locale.US);
+
+            if(!"en".equals(lang)) { // Only english is supported
+                throw new AnalysisEngineProcessException(MESSAGE_DIGEST,
+           							"unsupported_language_error", new Object[] { lang });
+            }
+
+            StemAnnotator stemAnnotator = new StemAnnotator(jcas);
+
+            for (String path : paths) {
+
+                // Separate Typename and featurepath
+                String[] segments = path.split("/", 2);
+                String typeName = segments[0];
+
+                // Try to get the type from the typesystem of the CAS
+                Type t = currCAS.getTypeSystem().getType(typeName);
+                if (t == null) {
+                    throw new IllegalStateException("Type [" + typeName + "] not found in type system");
+                }
+
+                // get an fpi object and initialize it
+                // initialize the FeaturePathInfo with the corresponding part
+                initializeFeaturePathInfoFrom(fp, segments);
+
+                // get the annotations
+                AnnotationIndex<?> idx = currCAS.getAnnotationIndex(t);
+                FSIterator<?> iterator = idx.iterator();
+
+                while (iterator.hasNext()) {
+                    AnnotationFS fs = (AnnotationFS) iterator.next();
+
+                    try {
+                        if (this.filterFeaturePath != null) {
+                            // check annotation filter condition
+                            if (this.filterFeaturePathInfo.match(fs, this.filterCondition)) {
+                                stemAnnotator.createStemAnnotation(fs);
+                            }
+                        }
+                        else { // no annotation filter specified
+                            stemAnnotator.createStemAnnotation(fs);
+                        }
+                    }
+                    catch (AnalysisEngineProcessException e) {
```

Comment:
Catching AnalysisEngineProcessException despite AnalysisEngineProcessException being declared in throws. This catch can be removed.